### PR TITLE
EZP-27901: As an Editor, I want to edit a DateAndTime field type with the new UI guidelines

### PR DESCRIPTION
**Pull request by sunpietro**
> JIRA: https://jira.ez.no/browse/EZP-27901
> Requires: https://github.com/ezsystems/repository-forms/pull/171

**THINGS DONE:**
- [x] implemented frontend part,
- [x] implemented backend part.

### CODE REVIEW

**Comment by webhdx** [comment]
Backend part is now finished: https://github.com/ezsystems/repository-forms/pull/171

File: src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        ],
+    });
+
+    validator.init();
+
+    const datetimeFields = [...document.querySelectorAll(SELECTOR_FIELD)];
+    const datetimeConfig = {
+        defaultDate: new Date(),
+        enableTime: true,
+        enableSeconds: true,
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js]: `enableSeconds` should come from backend. In fieldType you can specify if field should use seconds.

File: src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        sourceInput.dispatchEvent(new CustomEvent(EVENT_VALUE_CHANGED));
+    };
+    const initFlatPickr = (field) => {
+        const sourceInput = field.querySelector(SELECTOR_INPUT);
+        const pickrInput = document.createElement('input');
+
+        pickrInput.type = 'text';
+        pickrInput.classList.add('flatpickr', 'flatpickr-input', 'ez-data-source__input', 'form-control');
+
+        sourceInput.hidden = true;
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js]: Could this be done in backend while rendering field? //cc @webhdx

File: src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        sourceInput.parentNode.insertBefore(pickrInput, sourceInput);
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js]: Well this whole rendering additional input should be on backend in twig I think

File: src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                errorNodeSelectors: ['.ez-field-edit__label-wrapper'],
+            },
+        ],
+    });
+
+    validator.init();
+
+    const datetimeFields = [...document.querySelectorAll(SELECTOR_FIELD)];
+    const datetimeConfig = {
+        defaultDate: new Date(),
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js]: What about edit mode or when the field will have default value?

**Comment by sunpietro** [comment]
@dew326 The edit mode has been implemented. Ready for testing @micszo

**Comment by sunpietro** [comment]
@dew326 The edit mode has been implemented. Ready for testing @micszo

**Comment by sunpietro** [comment]
@m-tyrala the date/datetime/time field doesn't work this way. They don't use focus at all.
I'm not sure your request is valid.

**Comment by m-tyrala** [comment]
Here is what is written inside JIRA ticket:

> Validations
> The UI should validate the following scenario
> Required field
> The validation will be as it is on v1 when the field loses focus.

So i requested to change this, if this description is wrong, then no problem.

**Comment by lserwatka** [comment]
@m-tyrala you are right, could you make a bug in JIRA as a followup and paste is here?

**Comment by m-tyrala** [comment]
[EZP-28252](https://jira.ez.no/browse/EZP-28252)

**Pull request merged**